docs/recap/bitvm2_fast_recap.tex

```diff
@@ -0,0 +1,229 @@
+% Created 2024-09-23 Mon 13:35
+% Intended LaTeX compiler: pdflatex
+%%% TeX-command-extra-options: "-shell-escape"
+\documentclass[11pt]{article}
+\usepackage[utf8]{inputenc}
+\usepackage[T1]{fontenc}
+\usepackage{graphicx}
+\usepackage{longtable}
+\usepackage{wrapfig}
+\usepackage{rotating}
+\usepackage[normalem]{ulem}
+\usepackage{amsmath}
+\usepackage{amssymb}
+\usepackage{capt-of}
+\usepackage{hyperref}
+\makeatletter
+\newcommand{\citeprocitem}[2]{\hyper@linkstart{cite}{citeproc_bib_item_#1}#2\hyper@linkend}
+\makeatother
+\usepackage[a4paper, total={6in, 8in}]{geometry}
+\author{Kyrylo Baibula}
+\date{\today}
+\title{BitVM fast recap}
+\hypersetup{
+  pdfauthor={Kyrylo Baibula},
+  pdftitle={BitVM fast recap},
+  pdfkeywords={},
+  pdfsubject={},
+  pdfcreator={Emacs 29.4 (Org mode 9.7.11)},
+pdflang={English}}
+\usepackage{biblatex}
+
+\addbibresource{refs.bib}
+
+\begin{document}
+
+\maketitle
+
+\begin{abstract}
+  Assume some user wants to publicly execute a large program on the
+  Bitcoin chain using Bitcoin Script, but its implementation is larger
+  than 4 megabytes (for example, a Zero Knowledge proof); that is why it
+  cannot fit in the transaction input. This document is a fast recap of
+  the BitVM2 doc, which proposes optimistic execution verification with
+  fraud proofs in case of incorrectness.
+\end{abstract}
+
+\setcounter{tocdepth}{2}
+\tableofcontents
+
+\section{Introduction}\label{sec:intro}
+
+In the ever-evolving landscape of blockchain technology, the desire to
+execute increasingly complex and large-scale programs on the
+Bitcoin\autocite{bitcoin_paper} network is growing. However, Bitcoin
+Script, the native programming language of Bitcoin, imposes strict
+size limits, such as the 4-megabyte cap on transaction inputs, which
+makes it challenging to implement certain advanced cryptographic
+proofs, like Zero Knowledge proofs. To address this limitation, the
+BitVM2\autocite{bitvm2} proposal introduces an innovative approach
+that enables the optimistic execution of large programs on the Bitcoin
+chain. This method leverages fraud proofs to ensure the validity of
+executions, offering a robust mechanism for verifying computations,
+while providing a fail-safe against incorrect or malicious
+operations. This document provides a concise overview of BitVM2,
+highlighting its potential to unlock new possibilities for secure and
+scalable program execution on Bitcoin.
+
+\section{Program splitting}\label{sec:program-splitting}
+
+\begin{quote}
+  Let there be a large program \(f\), which can be described in Bitcoin
+  Script. We want to compute it \textbf{\textbf{on-chain}}, i.e., find
```

Comment:
Well, we are not trying to compute it on-chain, but rather **verify** some complex logic. In other words, we do the computation **off-chain**, but claim to compute it correctly **on-chain**

Reply:
Verifying is frequently the same as on-chain computation, but not necessarily

Reply:
Maybe let's then write it as: "verify the correct computation of some complex logic on-chain"